packages/core/src/sso/SamlSsoConnector/index.ts

```diff
@@ -0,0 +1,133 @@
+import { ConnectorError, ConnectorErrorCodes } from '@logto/connector-kit';
+import { type SsoConnector } from '@logto/schemas';
+import { assert } from '@silverhand/essentials';
+import * as saml from 'samlify';
+import { z } from 'zod';
+
+import SamlConnector, {
+  attributeMappingPostProcessor,
+  getUserInfoFromRawUserProfile,
+} from '../SamlConnector/index.js';
+import { type SingleSignOnFactory } from '../index.js';
+import { type SingleSignOn, SsoProviderName } from '../types/index.js';
+import { samlConnectorConfigGuard } from '../types/saml.js';
+
+/**
+ * SAML SSO connector
+ *
+ * This class extends the basic SAML connector class and add some business related utils methods.
+ *
+ * @property data The SAML connector data from the database
+ *
+ * @method getConfig Get parsed SAML config along with it's metadata. Throws error if config is invalid.
+ * @method getAuthorizationUrl Get the SAML auth URL.
+ * @method getUserInfo Get user info from the SAML assertion.
+ */
+export class SamlSsoConnector extends SamlConnector implements SingleSignOn {
+  constructor(
+    readonly data: SsoConnector,
+    tenantId: string
+  ) {
+    const parseConfigResult = samlConnectorConfigGuard.safeParse(data.config);
+
+    if (!parseConfigResult.success) {
+      throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, parseConfigResult.error);
+    }
+
+    super(parseConfigResult.data, tenantId, data.id);
+  }
+
+  /**
+   * Get parsed SAML connector's config along with it's metadata. Throws error if config is invalid.
+   *
+   * @returns Parsed SAML connector config and it's metadata.
+   */
+  async getConfig() {
+    return this.getSamlConfig();
+  }
+
+  /**
+   * Get the SAML auth URL.
+   *
+   * @returns The SAML auth URL.
+   */
+  async getAuthorizationUrl() {
```

Comment:
Why can we use the SamlConnector's `getSingleSignOn` method here?  All the metadata parsing logic should be placed within the `SamlConnector's` class?